pkg/api/v1/schema.go

```diff
@@ -72,12 +62,20 @@ type Disk struct {
 
 type ReadOnly struct{}
 
+// This generally follows style of k8s VolumeSource struct
 type DiskSource struct {
-	File          string          `json:"file,omitempty"`
-	StartupPolicy string          `json:"startupPolicy,omitempty"`
-	Protocol      string          `json:"protocol,omitempty"`
-	Name          string          `json:"name,omitempty"`
-	Host          *DiskSourceHost `json:"host,omitempty"`
+	PersistentVolumeClaim *DiskSourcePersistentVolumeClaim `json:"persistentVolumeClaim,omitempty"`
+	ISCSI                 *DiskSourceISCSI                 `json:"iscsi,omitempty"`
```

Comment:
This is designed to match the way POD volumes are configured - the POD volume source allows either referencing a PVC, or referencing an ISCSI server directly.

Since we've not figured out any way to make raw block device access work that, I'm explicitly ignoring that. This is only trying to deal with things that we can make work today.